bindings/netframework/ElectionGuard.NetFramework/ElectionGuardInterop/Ballot.h

```diff
@@ -1,4 +1,413 @@
+
+          bool IsValid(String ^ expectedObjectId)
+        {
+            std::string _expectedObjectId;
+            Utilities::MarshalString(expectedObjectId, _expectedObjectId);
+            return this->_instance->isValid(_expectedObjectId);
+        }
+    };
+
+  public
+    ref class CiphertextBallotSelection : ManagedInstance<electionguard::CiphertextBallotSelection>
+    {
+      public:
+        CiphertextBallotSelection(String ^ objectId, ElementModQ ^ descriptionHash,
+                                  ElGamalCiphertext ^ ciphertext, bool isPlaceholder,
+                                  ElementModQ ^ nonce, ElementModQ ^ cryptoHash,
+                                  DisjunctiveChaumPedersenProof ^ proof,
+                                  ElGamalCiphertext ^ extendedData)
+        {
+            std::string _objectId;
+            Utilities::MarshalString(objectId, _objectId);
+
+            this->_instance = new electionguard::CiphertextBallotSelection(
+              _objectId, *descriptionHash->_instance,
+              std::make_unique<electionguard::ElGamalCiphertext>(*ciphertext->_instance),
+              isPlaceholder, std::make_unique<electionguard::ElementModQ>(*nonce->_instance),
+              std::make_unique<electionguard::ElementModQ>(*cryptoHash->_instance),
+              std::make_unique<electionguard::DisjunctiveChaumPedersenProof>(*proof->_instance),
+              std::make_unique<electionguard::ElGamalCiphertext>(*extendedData->_instance));
+        }
+        internal : CiphertextBallotSelection(
+                     std::unique_ptr<electionguard::CiphertextBallotSelection> other)
+        {
+            this->_instance = other.release();
+        }
+
+      public:
+        property String ^ ObjectId {
+            String ^ get() {
+                auto value = this->_instance->getObjectId();
+                return gcnew String(value.c_str());
+            }
+        }
+    };
+
+  public
+    ref class PlaintextBallotContest : ManagedInstance<electionguard::PlaintextBallotContest>
+    {
+      public:
+        PlaintextBallotContest(String ^ objectId, array<PlaintextBallotSelection ^> ^ selections)
+        {
+            std::string _objectId;
+            Utilities::MarshalString(objectId, _objectId);
+
+            std::vector<std::unique_ptr<electionguard::PlaintextBallotSelection>> elements;
+
+            for each (auto item in selections) {
+                elements.push_back(item->_instance->clone());
+            }
+
+            this->_instance =
+              new electionguard::PlaintextBallotContest(_objectId, std::move(elements));
+        }
+        internal
+            : PlaintextBallotContest(std::unique_ptr<electionguard::PlaintextBallotContest> other)
+        {
+            this->_instance = other.release();
+        }
+        PlaintextBallotContest(electionguard::PlaintextBallotContest *unowned)
```

Comment:
similarly, for some objects, it is intended for the managed consumers to simply be able to read/access data members, so we can just hold a pointer without having to do a full copy operation.